bin/impl/deploy.sh

```diff
@@ -29,14 +29,33 @@ rm -rf $HADOOP_PREFIX/logs/application_*
 # Remove old deployment
 rm -rf $FLUO_HOME
 
-# Remove old tarball
-TARBALL=$FLUO_REPO/modules/distribution/target/fluo-$FLUO_VERSION-bin.tar.gz
-rm -f $TARBALL
-
-echo "Rebuilding Fluo" 
-# Create new tarball
-cd $FLUO_REPO
-mvn clean package -DskipTests -Daccumulo.version=$ACCUMULO_VERSION -Dhadoop.version=$HADOOP_VERSION
+if [ -n "$FLUO_TARBALL_PATH" ]; then
+  TARBALL=$FLUO_TARBALL_PATH
+elif [ -n "$FLUO_TARBALL_REPO" ]; then
+  # Remove old tarball
+  TARBALL=$FLUO_TARBALL_REPO/modules/distribution/target/fluo-$FLUO_VERSION-bin.tar.gz
+  rm -f $TARBALL
+
+  echo "Rebuilding Fluo" 
+  # Create new tarball
+  cd $FLUO_TARBALL_REPO
+  mvn clean package -DskipTests -Daccumulo.version=$ACCUMULO_VERSION -Dhadoop.version=$HADOOP_VERSION
+
+  if [ ! -f $TARBALL ]; then
+    echo "The tarball $TARBALL does not exist after building from the FLUO_TARBALL_REPO=$FLUO_TARBALL_REPO"
+    echo "Does your repo contain code matching the FLUO_VERSION=$FLUO_VERSION set in env.sh?"
+    exit 1
+  fi
+  
+else
+  TARBALL_FILENAME=fluo-distribution-$FLUO_VERSION-bin.tar.gz
+  if [ -f "$DOWNLOADS/$TARBALL_FILENAME" ]; then
+    echo "$TARBALL_FILENAME already exists in downloads/"
+  else
+    wget -P $DOWNLOADS $FLUO_TARBALL_URL
```

Comment:
I was writing another script that would donwload the fluo tar ball from GH.   During testing of that script I was repeatedly rerunnnig wget.  Eventually GH gave me a 404 instead of downloading it.    Not sure what what was going on.